lib/modules/dosomething/dosomething_kudos/dosomething_kudos.module

```diff
@@ -233,3 +234,62 @@ function dosomething_kudos_get_kudos_for_user_by_file($fid) {
 
   return $kudos;
 }
+
+/**
+ * Check to see if the kudos term has been selected by the user on specified
+ * reportback.
+ *
+ * @param  [type] $id
+ * @param  object $reportbackItem
+ * @return bool
+ */
+function dosomething_kudos_term_is_selected($id, $reportbackItem) {
+  if (! $reportbackItem->existing_kids) {
+    return false;
+  }
+
+  if (! isset($reportbackItem->existing_kids[$id])) {
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Get the icon associated with the kudos term id.
+ *
+ * @param  string $id
+ * @return string
+ * @TODO: this will eventually be changed to refere to SVG icon paths for our custom kudos icons!
+ */
+function dosomething_kudos_get_icon_by_term_id($id) {
+  settype($id, 'string');
+
+  $icons = [
+    '641' => '&#128150;',
+    '646' => '&#128169;',
+  ];
+
+  return $icons[$id];
+}
+
+/**
+ * Get the count for a kudos term on a Reportback item by the
+ * kudos term id.
+ *
+ * @param  string $id
+ * @param  object $reportbackItem
+ * @return string
+ */
+function dosomething_kudos_get_count_by_term_id($id, $reportbackItem) {
```

Comment:
we shouldn't need this function if we can call `dosomething_kudos_get_total_for_file_by_term_name`